homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from "react";
+import { Pillar } from "./components/Pillar/Pillar";
+import { arrayWithRandomValues } from "./utils/arrayWithRandomValues";
+import { bubbleSort, Sorted } from "./utils/bubbleSort";
+import "./App.css";
+import "./components/Pillar/pillar.css";
+
+interface AppProps {}
+interface AppState {
+  sortingArray: number[];
+  status: string;
+  delay: number;
+  alreadySorted?: boolean;
+}
+
+export class App extends Component<AppProps, AppState> {
+  interval: number | undefined;
+
+  state = {
+    sortingArray: arrayWithRandomValues(),
+    status: "pending",
+    delay: 100,
+    alreadySorted: Sorted,
+  };
+
+  sorting = (): void => {
+    this.interval = window.setInterval(() => {
+      if (this.state.alreadySorted) {
+        this.setState({ status: "SORTED" });
+        clearInterval(this.interval);
+      } else {
+        this.setState({
+          sortingArray: bubbleSort(this.state.sortingArray),
+          status: "sorting",
+          alreadySorted: Sorted,
+        });
+      }
+    }, this.state.delay);
+  };
+
+  pause = (): void => {
+    if (this.interval) {
+      clearInterval(this.interval);
+      this.setState(() => ({ status: "pause" }));
```

Comment:
Why do you pass a function here?

I would get rid of `setState` here. Because you reuse `pause` a lot of times and reset `status`. You don't need `setState` in all places where you reuse `pause`.

Consider moving reusable logic without `setState` in a separate method, for example.